src/components/progress-bar/progress-bar.scss

```diff
@@ -0,0 +1,282 @@
+@import "variables";
+@import "default-theme";
+
+$md-progress-bar-height: 5px !default;
+$md-progress-bar-full-animation-duration: 2s !default;
+$md-progress-bar-piece-animation-duration: 250ms !default;
+
+/** In buffer mode a repeated SVG object is used as a background.  Each of the following defines the SVG object for each
+    of the class defined colors.
+
+    Each string is a URL encoded version of:
+
+    <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
+         version="1.1" x="0px" y="0px" enable-background="new 0 0 5 2"
+         xml:space="preserve" viewBox="0 0 5 2" preserveAspectRatio="none slice">
+      <circle cx="1" cy="1" r="1" fill="{INJECTED_COLOR}"/>
+    </svg>
+
+    */
+$md-buffer-bubbles-primary: (
+  "data:image/svg+xml;charset=UTF-8,%3Csvg%20version%3D%271.1%27%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27" +
+  "%20xmlns%3Axlink%3D%27http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%27%20x%3D%270px%27%20y%3D%270px%27%20enable-backgroun" +
+  "d%3D%27new%200%200%205%202%27%20xml%3Aspace%3D%27preserve%27%20viewBox%3D%270%200%205%202%27%20preserveAspectRatio" +
+  "%3D%27none%20slice%27%3E%3Ccircle%20cx%3D%271%27%20cy%3D%271%27%20r%3D%271%27%20fill%3D%27" +
+  md-color($md-primary, 100) + "%27%2F%3E%3C%2Fsvg%3E") !default;
+$md-buffer-bubbles-accent: (
+  "data:image/svg+xml;charset=UTF-8,%3Csvg%20version%3D%271.1%27%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27" +
+  "%20xmlns%3Axlink%3D%27http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%27%20x%3D%270px%27%20y%3D%270px%27%20enable-backgroun" +
+  "d%3D%27new%200%200%205%202%27%20xml%3Aspace%3D%27preserve%27%20viewBox%3D%270%200%205%202%27%20preserveAspectRatio" +
+  "%3D%27none%20slice%27%3E%3Ccircle%20cx%3D%271%27%20cy%3D%271%27%20r%3D%271%27%20fill%3D%27" +
+  md-color($md-accent, 100) + "%27%2F%3E%3C%2Fsvg%3E") !default;
+$md-buffer-bubbles-warn: (
+  "data:image/svg+xml;charset=UTF-8,%3Csvg%20version%3D%271.1%27%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27" +
+  "%20xmlns%3Axlink%3D%27http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%27%20x%3D%270px%27%20y%3D%270px%27%20enable-backgroun" +
+  "d%3D%27new%200%200%205%202%27%20xml%3Aspace%3D%27preserve%27%20viewBox%3D%270%200%205%202%27%20preserveAspectRatio" +
+  "%3D%27none%20slice%27%3E%3Ccircle%20cx%3D%271%27%20cy%3D%271%27%20r%3D%271%27%20fill%3D%27" +
+  md-color($md-warn, 100) + "%27%2F%3E%3C%2Fsvg%3E") !default;
+
+:host {
+  display: block;
+  /** Height is provided for md-progress-bar to act as a default. */
+  height: $md-progress-bar-height;
+  overflow: hidden;
+  position: relative;
+  /** translateZ is added to force the md-progress-bar into its own GPU layer. */
+  transform: translateZ(0);
+  transition: opacity $md-progress-bar-piece-animation-duration linear;
+  width: 100%;
+
+  .background {
```

Comment:
Even if nesting is okay, I still want to prefix these classes so that it's clear in the dev tools where they come from.